engine-plugins/identity-ldap/src/main/java/org/camunda/bpm/identity/impl/ldap/LdapIdentityProviderSession.java

```diff
@@ -336,14 +339,19 @@ public boolean checkPassword(String userId, String password) {
       return false;
     } else {
 
+      LdapContext context = null;
+
       try {
         // bind authenticate for user + supplied password
-        openContext(user.getDn(), password);
+        context = openContext(user.getDn(), password);
         return true;
 
       } catch(LdapAuthenticationException e) {
         return false;
 
+      } finally {
+        closeLdapCtx(context);
+
       }
```

Comment:
Can we avoid reassigning the `initialContext` and instead close the context immediately that was created to check the password?

Reply:
Seems ok for me. 
The CommandContext.closeSession() will try to close it again. Should'nt really matter as far as i can see.

Reply:
The method `CommandContext#closeSession` tries to close the `initialContext` again. This is the same behavior as before, right? Or am I missing something?

Reply:
Yes that's right. Your changes should be fine